lib/hanami/helpers/form_helper/form_builder.rb

```diff
@@ -766,6 +766,45 @@ def select(name, values, attributes = {})
           end
         end
 
+        # Datalist input
+        #
+        # @param name [Symbol] the input name
+        # @param values [Hash] a Hash to generate <tt><option></tt> tags of datalist.
+        #   Keys correspond to <tt>value</tt> and values correspond to the content.
+        # @param list [String] the name of list for the text input, it's also the id of datalist
+        # @param attributes [Hash] HTML attributes to pass to the input tag
+        #
+        # @since 0.3.1
+        #
+        # @example Basic usage
+        #   <%=
+        #     # ...
+        #     values = Hash['it' => 'Italy', 'us' => 'United States']
+        #     datalist :stores, values, 'books'
+        #   %>
+        #
+        #   # Output:
+        #   #  <input type="text" name="book[store]" id="book-store" list="books">
+        #   #  <datalist id="books">
+        #   #    <option value="Italy"></option>
+        #   #    <option value="United States"></option>
+        #   #  </datalist>
+        def datalist(name, values, list, attributes = {})
+          options  = attributes.delete(:options) || {}
+          datalist = attributes.delete(:datalist) || {}
+          attributes = { type: 'text', name: _input_name(name), id: _input_id(name), list: list }.merge(attributes)
+
+          # remove 'id' attribute
+          datalist = { id: list }.merge(datalist)
+
+          input(attributes)
+          super(datalist) do
+            values.each do |value, content|
+              option(value, {value: content}.merge(options))
```

Comment:
yes, @joneslee85 right. I want to say that this code:
``` ruby
options[:value] = content
option(value, options)
```
will be faster than current solution.